### PR TITLE
feat: [CO-1343] Add LDAP module config update script

### PR DESCRIPTION
**Pull request by keshavbhatt**
This script will run during the post-installation hook of the **carbonio-directory-server** package upon upgrades. 

Its purpose is to **ensure that the LDAP module configuration is updated whenever modifications are applied to the modules loaded by carbonio-openldap**.

The decision to include this script here, rather than directly in the carbonio-directory-server package, is based on historical reasons. Additionally, separating this script helps prevent code duplication across different distribution-specific post-installation instructions in the PKGBUILD file.

**Linked PRs:**
- https://github.com/zextras/carbonio-mailbox/pull/548
- https://github.com/zextras/carbonio-core/pull/75

### CODE REVIEW

**Comment by keshavbhatt** [comment]
@M0Rf30 I would like to backup the target file before performing replacement. What is the ideal location in your opinion to store this file. I would avoid backing it up in the same directory cause slapd own and use this directory.

**Comment by keshavbhatt** [comment]
> @M0Rf30 I would like to backup the target file before performing replacement. What is the ideal location in your opinion to store this file. I would avoid backing it up in the same directory cause slapd own and use this directory.

I decided to use `/opt/zextras/data/ldap-post-install-backup/` as backup directory. The script will take backup before replacing the original file.

**Review by aheeva-yuliya — APPROVED** [review]

**Review by sonersivri — APPROVED** [review]

**Comment by frisonisland** [comment]
@keshavbhatt 
Just a question: I know there are some old migrations scripts that can achieve similar results, what about suing this https://github.com/zextras/carbonio-core-utils/blob/devel/src/libexec/zmldapapplyldif ? (I'm not even sure we are using it)

I'm asking because currently Ldap is updated with very different procedures, and I think having just one procedure could be easier to manage and maintain. 
For ldif files, in general, we can also do a simple "ldapModify -f <file.ldif>". What do you think about it?

**Comment by keshavbhatt** [comment]
@frisonisland some configuration changes like this one (module update) cannot be achieved with applying OLC updates. I mentioned these details in ticket. To apply OLC updates we need to ensure the server is up and running.  The serve in this case is refusing to start cause it cannot load the module with was removed by new package update.

**Review by frisonisland — APPROVED** [review]

File: src/libexec/ldapmoduleconfupdate [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        echo "Error: Target module config file $TARGET_MODULE_CONF_FILE does not exist."
+        exit 1
+    fi
+
+    if [ ! -f "$SOURCE_MODULE_CONF_FILE" ]; then
+        echo "Error: Source module config file $SOURCE_MODULE_CONF_FILE does not exist."
+        exit 1
+    fi
+
+    if ! diff "$TARGET_MODULE_CONF_FILE" "$SOURCE_MODULE_CONF_FILE" > /dev/null; then
```
Comment (M0Rf30) [inline src/libexec/ldapmoduleconfupdate:36]: `diff` requires `diffutils` package installed. is there another way to achieve this, with more common and already used commands?

**Review by M0Rf30 — CHANGES_REQUESTED** [review]
Indent with `shfmt -i 2 -ci`

File: src/libexec/ldapmoduleconfupdate (hunk 1 above)
Comment (M0Rf30) [reply src/libexec/ldapmoduleconfupdate]: also `cmp` is a diffutils binary

Comment (keshavbhatt) [reply src/libexec/ldapmoduleconfupdate]: And the `diffutils` do not come preinstalled on our supported distributions? Do you have any other suggestions @M0Rf30

Comment (M0Rf30) [reply src/libexec/ldapmoduleconfupdate]: md5sum, it's part of coreutils

Comment (keshavbhatt) [reply src/libexec/ldapmoduleconfupdate]: @M0Rf30 I added `diffutils` in `carbonio-core` as dependency. We should be fine using `cmp` now. Can you review the changes?

**Review by M0Rf30 — APPROVED** [review]

**Pull request merged**